app/scripts/metamask-controller.js

```diff
@@ -4266,6 +4272,73 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Adds a new mnemonic to the vault.
+   *
+   * @param {string} mnemonic
+   * @returns {object} new account address
+   */
+  ///: BEGIN:ONLY_INCLUDE_IF(multi-srp)
+  async addNewMnemonicToVault(mnemonic) {
+    const releaseLock = await this.createVaultMutex.acquire();
+    try {
+      // TODO: remove and use this logic in the keyring controller.
+      const findMostRecentIndex = (arr, targetType) => {
+        return arr.reduceRight((acc, item, index) => {
+          if (acc === -1 && item.type === targetType) {
+            return index;
+          }
+          return acc;
+        }, -1);
+      };
+
+      const newKeyring = await this.keyringController.addNewKeyring(
+        KeyringTypes.hd,
+        {
+          mnemonic,
+          numberOfAccounts: 1,
+        },
+      );
+      const [newAccountAddress] = await newKeyring.getAccounts();
+      const account =
+        this.accountsController.getAccountByAddress(newAccountAddress);
+      this.accountsController.setSelectedAccount(account.id);
+
+      // TODO: Find a way to encapsulate this logic in the KeyringController itself.
+      const mostRecentHDKeyringIndex = findMostRecentIndex(
+        this.keyringController.state.keyrings,
+        KeyringTypes.hd,
+      );
+      const keyringId =
+        this.keyringController.state.keyringsMetadata[mostRecentHDKeyringIndex]
+          .id;
```

Comment:
Follow-up of my other comment:

```suggestion
      const keyringId = findKeyringIdByAddress(newAccountAddress);
```